aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -55,18 +61,29 @@ void atan2_kernel(TensorIteratorBase& iter) {
 }
 
 void mul_kernel(TensorIteratorBase& iter) {
-  if (iter.dtype() == ScalarType::Bool) {
+  auto dtype = iter.common_dtype();
+  if (dtype == ScalarType::Bool) {
     cpu_kernel(iter, [=](bool a, bool b) -> bool { return a && b; });
-  } else if (iter.dtype() == kComplexHalf) {
+  } else if (dtype == kComplexHalf) {
     cpu_kernel(
         iter,
         [=](c10::complex<at::Half> a,
             c10::complex<at::Half> b) -> c10::complex<at::Half> {
           using comp_t = c10::complex<float>;
           return comp_t{a} * comp_t{b};
         });
+  } else if (iter.is_scalar(2) && at::isReducedFloatingType(dtype)) {
```

Comment:
why is it only `is_scalar(2)`, is `mul` constructed in such a way that scalar argument is the second?

Reply:
TensorIterator has logic to purmute scalar input to the rear, so this patch will work for both of the following senarios:
```
>>> torch.tensor([0.01], dtype=torch.float16) * 1e5
tensor([1000.], dtype=torch.float16)
>>> 1e5 * torch.tensor([0.01], dtype=torch.float16)
tensor([1000.], dtype=torch.float16)
```